redis/src/commands/mod.rs

```diff
@@ -1257,6 +1263,157 @@ implement_commands! {
         cmd("ZUNIONSTORE").arg(dstkey).arg(keys.num_of_args()).arg(keys).arg("AGGREGATE").arg("MAX").arg("WEIGHTS").arg(weights)
     }
 
+    // vector set commands
+
+    /// Add a new element into the vector set specified by key.
+    /// [Redis Docs](https://redis.io/commands/VADD)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vadd<K: ToRedisArgs, E: ToRedisArgs>(key: K, input: VectorAddInput<'a>, element: E) -> (bool) {
+        cmd("VADD").arg(key).arg(input).arg(element)
+    }
+
+    /// Add a new element into the vector set specified by key with optional parameters for fine-tuning the insertion process.
+    /// [Redis Docs](https://redis.io/commands/VADD)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vadd_extended<K: ToRedisArgs, E: ToRedisArgs>(key: K, input: VectorAddInput<'a>, element: E, options: &'a VAddOptions) -> (bool) {
+        cmd("VADD").arg(key).arg(options.reduction_dimension.map(|_| "REDUCE")).arg(options.reduction_dimension).arg(input).arg(element).arg(options)
+    }
+
+    /// Get the number of members in a vector set.
+    /// [Redis Docs](https://redis.io/commands/VCARD)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vcard<K: ToRedisArgs>(key: K) -> (usize) {
+        cmd("VCARD").arg(key)
+    }
+
+    /// Return the number of dimensions of the vectors in the specified vector set.
+    /// [Redis Docs](https://redis.io/commands/VDIM)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vdim<K: ToRedisArgs>(key: K) -> (usize) {
+        cmd("VDIM").arg(key)
+    }
+
+    /// Return the approximate vector associated with a given element in the vector set.
+    /// [Redis Docs](https://redis.io/commands/VEMB)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vemb<K: ToRedisArgs, E: ToRedisArgs>(key: K, element: E) -> Generic {
+        cmd("VEMB").arg(key).arg(element)
+    }
+
+    /// Return the raw internal representation of the approximate vector associated with a given element in the vector set.
+    /// Vector sets normalize and may quantize vectors on insertion.
+    /// VEMB reverses this process to approximate the original vector by de-normalizing and de-quantizing it.
+    /// [Redis Docs](https://redis.io/commands/VEMB)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vemb_raw<K: ToRedisArgs, E: ToRedisArgs>(key: K, element: E) -> Generic {
```

Comment:
it's preferable to use an option type here, in case that more values will be added to the command in the future.

Reply:
This aligns with the established convention of using a separate function definition to add a simple parameter such as "RAW". The pattern can be observed in the names of functions that use "WITHSCORES" as a parameter. As of now this is the only variable to the VEMB command and this influenced the decision not to wrap it in an option type. With that being said, I don't mind changing it if you think it is necessary.

Reply:
Done